pallets/dapp-staking-migration/src/lib.rs

```diff
@@ -0,0 +1,692 @@
+                    Err(<T as Config>::WeightInfo::migrate_ledger_noop())
+                }
+            }
+        }
+
+        /// Used to remove one entry from the old _dapps_staking_v2_ storage.
+        ///
+        /// If there are no more entries to remove, returns `Err(_)` with consumed weight and number of deleted entries.
+        /// Otherwise returns `Ok(_)` with consumed weight and number of consumed enries.
+        pub(crate) fn cleanup_old_storage(limit: u32) -> Result<(Weight, u32), (Weight, u32)> {
+            let hashed_prefix = twox_128(pallet_dapps_staking::Pallet::<T>::name().as_bytes());
+
+            // Repeated calls in the same block don't work, so we set the limit to `Unlimited` in case of `try-runtime` testing.
+            let inner_limit = if cfg!(feature = "try-runtime") {
+                None
+            } else {
+                Some(limit)
+            };
+
+            let (keys_removed, done) = match clear_prefix(&hashed_prefix, inner_limit) {
+                KillStorageResult::AllRemoved(value) => (value, true),
+                KillStorageResult::SomeRemaining(value) => (value, false),
+            };
+
+            log::trace!(
+                target: LOG_TARGET,
+                "Removed {} keys from storage.",
+                keys_removed
+            );
+
+            if !done {
+                Ok((
+                    <T as Config>::WeightInfo::cleanup_old_storage_success()
+                        .saturating_mul(keys_removed.into()),
+                    keys_removed as u32,
+                ))
+            } else {
+                log::trace!(target: LOG_TARGET, "All keys have been removed.",);
+                Err((
+                    <T as Config>::WeightInfo::cleanup_old_storage_noop(),
+                    keys_removed as u32,
+                ))
+            }
+        }
+
+        /// Max allowed weight that migration should be allowed to consume.
+        pub(crate) fn max_call_weight() -> Weight {
+            // 50% of block should be fine
+            T::BlockWeights::get().max_block / 2
+        }
+
+        /// Min allowed weight that migration should be allowed to consume.
+        ///
+        /// This serves as a safety marging, to prevent accidental overspending, due to
+        /// inprecision in implementation or benchmarks, when small weight limit is specified.
+        pub(crate) fn min_call_weight() -> Weight {
+            // 5% of block should be fine
+            T::BlockWeights::get().max_block / 10
+        }
+
+        /// Calculate call weight to use.
+        ///
+        /// In case of `None`, use the max allowed call weight.
+        /// Otherwise clamp the specified weight between the allowed min & max values.
+        fn clamp_call_weight(weight: Option<Weight>) -> Weight {
+            weight
+                .unwrap_or(Self::max_call_weight())
+                .min(Self::max_call_weight())
+                .max(Self::min_call_weight())
```

Comment:
Should it be ?
```
weight
    .unwrap_or(Self::max_call_weight())
    .min(Self::min_call_weight())
    .max(Self::max_call_weight())
```

Reply:
It can be confusing, but we can do a sanity check 🙂 

* min value = 5
* max value = 10

_1st scenario_
* 7 is provided
* 7.min(10).max(5) = 7.max(5) = 7;

_2nd scenario__
* 4 is provided
* 4.min(10).max(5) = 4.max(5) = 5;

_3rd scenario_
* 11 is provided
* 11.min(10).max(5) = 10.max(5) = 10